Sources/ENCore/App/Features/Network/NetworkController.swift

```diff
@@ -5,234 +5,212 @@
  *  SPDX-License-Identifier: EUPL-1.2
  */
 
-import Combine
 import ENFoundation
 import Foundation
-import Reachability
+import RxSwift
 
 final class NetworkController: NetworkControlling, Logging {
 
     // MARK: - Init
 
     init(networkManager: NetworkManaging,
-         cryptoUtility: CryptoUtility,
-         mutableNetworkStatusStream: MutableNetworkStatusStreaming) {
+         cryptoUtility: CryptoUtility) {
         self.networkManager = networkManager
         self.cryptoUtility = cryptoUtility
-        self.mutableNetworkStatusStream = mutableNetworkStatusStream
     }
 
     // MARK: - NetworkControlling
 
-    var applicationManifest: AnyPublisher<ApplicationManifest, NetworkError> {
-        return Deferred {
-            Future { promise in
-                self.networkManager.getManifest { result in
-                    promise(result.map { $0.asApplicationManifest })
+    var applicationManifest: Single<ApplicationManifest> {
+        let observable: Single<ApplicationManifest> = .create { [weak self] observer in
+            guard let strongSelf = self else {
+                observer(.failure(ExposureDataError.internalError))
+                return Disposables.create()
+            }
+
+            strongSelf.networkManager.getManifest { result in
+                switch result {
+                case let .failure(error):
+                    observer(.failure(error))
+                case let .success(manifest):
+                    observer(.success(manifest.asApplicationManifest))
                 }
             }
+            return Disposables.create()
         }
-        .receive(on: DispatchQueue.main)
-        .eraseToAnyPublisher()
+
+        return observable.observe(on: MainScheduler.instance)
```

Comment:
Here, the observable is scheduled to be observed on the Main queue, like in the combine implementation. For the requests below however, this has been removed. Is this intentional?

Reply:
I agree this is not very consistent. I will make an internal ticket to align this and make all network calls in this class return on the main thread.